src/Features/CSharp/Portable/InitializeParameter/InitializeParameterHelpers.cs

```diff
@@ -2,17 +2,63 @@
 
 using System.Collections.Immutable;
 using System.Diagnostics;
+using System.Threading;
 using Microsoft.CodeAnalysis.CSharp.Extensions;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.Editing;
 using Microsoft.CodeAnalysis.Operations;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CSharp.InitializeParameter
 {
     internal static class InitializeParameterHelpers
     {
-        public static SyntaxNode GetBody(BaseMethodDeclarationSyntax containingMember)
-            => containingMember.Body ?? (SyntaxNode)containingMember.ExpressionBody;
+        public static bool IsFunctionDeclaration(SyntaxNode node)
+            => node is BaseMethodDeclarationSyntax
+            || node is LocalFunctionStatementSyntax
+            || node is AnonymousFunctionExpressionSyntax;
+
+        public static IBlockOperation GetBlockOperation(SyntaxNode functionDeclaration, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            var bodyOpt = GetBody(functionDeclaration);
+            if (bodyOpt == null)
+                return null;
+
+            // body might be an expression in a lambda - in that case we wouldn't get a block operation out of that
+            return functionDeclaration is AnonymousFunctionExpressionSyntax
+                ? ((IAnonymousFunctionOperation)semanticModel.GetOperation(functionDeclaration, cancellationToken)).Body
+                : (IBlockOperation)semanticModel.GetOperation(bodyOpt, cancellationToken);
+        }
+
+        private static SyntaxNode GetBody(SyntaxNode functionDeclaration)
+        {
+            switch (functionDeclaration)
+            {
+                case BaseMethodDeclarationSyntax methodDeclaration:
+                    return (SyntaxNode)methodDeclaration.Body ?? methodDeclaration.ExpressionBody;
+                case LocalFunctionStatementSyntax localFunction:
+                    return (SyntaxNode)localFunction.Body ?? localFunction.ExpressionBody;
+                case AnonymousFunctionExpressionSyntax anonymousFunction:
+                    return (SyntaxNode)anonymousFunction.Body;
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(functionDeclaration);
+            }
+        }
+
+        private static SyntaxToken? GetSemicolonToken(SyntaxNode functionDeclaration)
```

Comment:
preference is to call this TryGetSemicolonToken to indicate that null is a reasonable value you can get back here.

Reply:
isn't that what the `?` is for?

Reply:
If you look at the signature yes.  But at the reference point it's not clear.  We find the code reads more clearly in certain cases if you write it as "TryXXX".  (And yes, this isn't consistently applied).

Reply:
done